### ISSUE TITLE
Feature request: 2FA in pre-boot authentication for FDE

### ISSUE THREAD

**Issue by pandapandapandapanda** [veracrypt/VeraCrypt#103]
I am helping a client set up FDE with Veracrypt and they are asking whether something like this would be possible to implement: to pass pre-OS authentication a user may be required not only to know the correct password, but also to have e.g. a USB flash drive with a specific hardware ID or a specific file in the root of the filesystem. Their argument is that if that was doable, that would be a kind of two-factor authentication for Veracrypt's FDE.

**Comment by idrassi** [comment]
Yes, VeraCrypt EFI bootloader is generic enough to implement a 2FA solution based on external hardware or even a TPM. Actually, there is already some code that allows blocking boot unless a USB with specific hardware ID is present but it is not finalized and more work is needed to make is usable and secure.

We are open to any help or sponsoring to implement this feature.

**Comment by andreas-becker** [comment]
@idrassi I think this issue can be closed, as the author does not exist anymore and you answered it already.

**Comment by MADXhh** [comment]
@idrassi 
Is there anything new on this topic? 
Do you already know when it will be possible?
Thanks!!
